3.6/debian-9/rootfs/opt/bitnami/scripts/libmongodb.sh

```diff
@@ -1078,8 +1078,25 @@ mongodb_is_file_external() {
 #   None
 #########################
 mongodb_custom_init_scripts() {
-    info "Loading custom scripts..."
-    if [[ -n $(find "$MONGODB_INITSCRIPTS_DIR/" -type f -regex ".*\.\(sh\|js\|js.gz\)") ]] && [[ ! -f "$MONGODB_VOLUME_DIR/.user_scripts_initialized" ]] ; then
+    local run_custom_init_scripts="no"
+    if [[ -n "$MONGODB_REPLICA_SET_MODE" ]]; then
+        if [[ "$MONGODB_REPLICA_SET_MODE" != "primary" ]]; then
+            debug "Skipping loading custom scripts on non-primary nodes..."
+        elif [[ -n $(find "$MONGODB_INITSCRIPTS_DIR/" -type f -regex ".*\.\(sh\|js\|js.gz\)") ]]; then
+            if [[ -f "$MONGODB_VOLUME_DIR/mongodb/.user_scripts_initialized" ]]; then
+                debug "Skippig loading custom scripts on container restarts..."
+            else
+                run_custom_init_scripts="yes"
+            fi
+        fi
+    elif [[ -n $(find "$MONGODB_INITSCRIPTS_DIR/" -type f -regex ".*\.\(sh\|js\|js.gz\)") ]]; then
+        if [[ -f "$MONGODB_VOLUME_DIR/mongodb/.user_scripts_initialized" ]]; then
+            debug "Skippig loading custom scripts on container restarts..."
+        else
+            run_custom_init_scripts="yes"
+        fi
+    fi
+    if is_boolean_yes "$run_custom_init_scripts"; then
```

Comment:
Why not just

```suggestion
    if [[ -n $(find "$MONGODB_INITSCRIPTS_DIR/" -type f -regex ".*\.\(sh\|js\|js.gz\)") ]] && [[ ! -f "$MONGODB_VOLUME_DIR/.user_scripts_initialized" ]] && ([[ -z "$MONGODB_REPLICA_SET_MODE" ]] || [[ "$MONGODB_REPLICA_SET_MODE" == "primary" ]]); then
```

It avoids duplicity in the conditions, no?

Reply:
So we can have a proper debug message when we detect the flag